source/extensions/common/wasm/wasm.cc

```diff
@@ -2127,20 +2128,35 @@ void Wasm::getFunctions() {
 }
 
 Wasm::Wasm(const Wasm& wasm, Event::Dispatcher& dispatcher)
-    : std::enable_shared_from_this<Wasm>(wasm), cluster_manager_(wasm.cluster_manager_),
-      dispatcher_(dispatcher), scope_(wasm.scope_), direction_(wasm.direction_),
-      local_info_(wasm.local_info_), listener_metadata_(wasm.listener_metadata_), id_(wasm.id_),
-      owned_scope_(wasm.owned_scope_), time_source_(dispatcher.timeSource()),
-      stat_name_set_(scope_.symbolTable()) {
+    : std::enable_shared_from_this<Wasm>(wasm), configured_vm_id_(wasm.configured_vm_id_),
+      vm_id_(wasm.vm_id_), creating_plugin_(wasm.creating_plugin_),
+      cluster_manager_(wasm.cluster_manager_), dispatcher_(dispatcher),
+      time_source_(dispatcher.timeSource()),
+      stat_name_set_(creating_plugin_->scope_.symbolTable()) {
   wasm_vm_ = wasm.wasmVm()->clone();
   vm_context_ = std::make_shared<Context>(this);
   getFunctions();
 }
 
-bool Wasm::initialize(const std::string& code, absl::string_view name, bool allow_precompiled) {
+bool Wasm::initialize(const std::string& code, bool allow_precompiled) {
   if (!wasm_vm_) {
     return false;
   }
+
+  // If the configured_vm_id is empty, then hash the code to create a unique vm_id.
+  if (vm_id_.empty()) {
+    std::vector<uint8_t> digest(SHA256_DIGEST_LENGTH);
```

Comment:
Please create a variant of `Envoy::Common::Crypto::Utility::getSha256Digest()` that takes `const std::string&` or `absl::string_view` as an argument and use it here instead of an inline implementation.

Reply:
Done.